lib/msal-browser/src/app/PublicClientApplication.ts

```diff
@@ -558,17 +554,54 @@ export class PublicClientApplication {
     }
 
     /**
-     * Generates a request that will contain the openid and profile scopes.
-     * @param request 
+     * Creates an Authorization Code Client with the given authority, or the default authority.
+     * @param authorityUrl 
      */
-    private generateLoginRequest(request: AuthorizationUrlRequest): AuthorizationUrlRequest {
-        const loginRequest = { ...request };
-        if (!loginRequest.scopes) {
-            loginRequest.scopes = [Constants.OPENID_SCOPE, Constants.PROFILE_SCOPE];
-        } else {
-            loginRequest.scopes.push(Constants.OPENID_SCOPE, Constants.PROFILE_SCOPE);
-        }
-        return loginRequest;
+    private async createAuthCodeClient(authorityUrl?: string): Promise<AuthorizationCodeClient> {
+        // Create auth module.
+        const clientConfig = await this.getClientConfiguration(authorityUrl);
+        return new AuthorizationCodeClient(clientConfig);
+    }
+
+    /**
+     * Creates an Silent Flow Client with the given authority, or the default authority.
+     * @param authorityUrl 
+     */
+    private async createSilentFlowClient(authorityUrl?: string): Promise<SilentFlowClient> {
+        // Create auth module.
+        const clientConfig = await this.getClientConfiguration(authorityUrl);
+        return new SilentFlowClient(clientConfig);
+    }
+
+    /**
+     * Creates a Client Configuration object with the given authority, or the default authority.
+     * @param authorityUri 
+     */
+    private async getClientConfiguration(authorityUri?: string): Promise<ClientConfiguration> {
+        return {
+            authOptions: {
+                clientId: this.config.auth.clientId,
+                authority: authorityUri ? await AuthorityFactory.createDiscoveredInstance(authorityUri, this.config.system.networkClient) : await this.defaultAuthorityPromise,
```

Comment:
Can you pull this into a local variable before the return statement? Would make it easier to read.

Reply:
done.